content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,315 @@
+This article is about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+
+### Introduction
+
+Socket programming is a means of communicating data between two computers across a network. Connections can be made using either a **connection-oriented protocol** or a **connectionless protocol**. TCP/IP, a connection-oriented protocol, will be used in our case.
+
+Before exchanging data, two computers must establish a link in connection-oriented protocols. In contrast, you don't require every packet to arrive at its intended destination for connectionless. It is made feasible via UDP (User Datagram Protocol).
```

Comment:
> In contrast, you don't require every packet to arrive at its intended destination for connectionless

I don't believe it is really clear for a beginner what you are talking about in this sentence. I would elaborate on this further. It doesn't have to be a long explanation since this article is focused on the connection-oriented protocol. I had to do research to understand what you were trying to say here which is not ideal

Reply:
> It is made feasible via UDP (User Datagram Protocol).

This sentence makes it sound like UDP is the only option for connectionless protocol. I did some research and it appears other protocols are connectionless protocols. Maybe say something like:

> UDP (User Datagram Protocol) is an example of a connectionless protocol where this is feasible.